ports/servoshell/prefs.rs

```diff
@@ -210,438 +213,333 @@ pub fn read_prefs_map(txt: &str) -> HashMap<String, PrefValue> {
+#[derive(Bpaf, Clone, Debug)]
+#[bpaf(options, version(VERSION))]
+struct CmdArgs {
+    /// Background Hang Monitor enabled.
+    #[bpaf(short('B'), long("bhm"))]
+    background_hang_monitor: bool,
+
+    /// Path to find SSL certificates.
+    #[bpaf(argument("/home/servo/resources/certs"))]
+    certificate_path: Option<PathBuf>,
+
+    /// Do not shutdown until all threads have finished (macos only).
+    #[bpaf(long)]
+    clean_shutdown: bool,
+
+    /// Config directory following xdg spec on linux platform.
+    #[bpaf(argument("~/.config/servo"))]
+    config_dir: Option<PathBuf>,
+
+    /// Run as a content process and connect to the given pipe.
+    #[bpaf(argument("servo-ipc-channel.abcdefg"))]
+    content_process: Option<String>,
+
+    /// A comma-separated string of debug options. Pass help to show available options.
+    #[bpaf(
+        short('Z'),
+        argument("layout_grid_enabled=true,dom_async_clipboard_enabled"),
+        long,
+        map(map_debug_options),
+        fallback(vec![])
+    )]
+    debug: Vec<String>,
+
+    /// Device pixels per px.
+    #[bpaf(argument("1.0"))]
+    device_pixel_ratio_override: Option<f32>,
+
+    /// Start remote devtools server on port.
+    #[bpaf(argument("0"))]
+    devtools: Option<u16>,
+
+    /// Whether or not to enable experimental web platform features.
+    #[bpaf(long)]
+    enable_experimental_web_platform_features: bool,
+
+    // Exit after Servo has loaded the page and detected a stable output image.
+    #[bpaf(short('x'), long)]
+    exit: bool,
+
+    // Use ipc_channel in singleprocess mode.
+    #[bpaf(short('I'), long("force-ipc"))]
+    force_ipc: bool,
+
+    /// Exit on thread failure instead of displaying about:failure.
+    #[bpaf(short('f'), long)]
+    hard_fail: bool,
+
+    /// Headless mode.
+    #[bpaf(short('z'), long)]
+    headless: bool,
+
+    /// Whether or not to completely ignore certificate errors.
+    #[bpaf(long)]
+    ignore_certificate_errors: bool,
+
+    /// Number of threads to use for layout.
+    #[bpaf(short('y'), long, argument("1"))]
+    layout_threads: Option<i64>,
+
+    /// Directory root with unminified scripts.
+    #[bpaf(argument("~/.local/share/servo"))]
+    local_script_source: Option<PathBuf>,
 
     #[cfg(target_env = "ohos")]
-    let log_filter = {
-        let filters = opt_match.opt_strs("log-filter").join(",");
-        let log_filter = (!filters.is_empty()).then_some(filters).or_else(|| {
-            (!preferences.log_filter.is_empty()).then_some(preferences.log_filter.clone())
-        });
-        log::debug!("Set log_filter to: {:?}", log_filter);
-        log_filter
-    };
+    /// Define a custom filter for logging.
+    #[bpaf(argument("FILTER"))]
+    log_filter: Option<String>,
 
     #[cfg(target_env = "ohos")]
-    let log_to_file = opt_match.opt_present("log-to-file");
+    /// Also log to a file (/data/app/el2/100/base/org.servo.servo/cache/servo.log).
+    #[bpaf(long, fallback(false))]
```

Comment:
switch gets automagical fallback value of `false`

```suggestion
    #[bpaf(long)]
```